src/core/reducers/carousels.js

```diff
@@ -0,0 +1,40 @@
+import { knuthShuffle } from 'knuth-shuffle';
+
+
+// Carousels have three items max, all the time :-)
+const MAX_ITEMS = 3;
+export const SET_CAROUSEL_ORDER = 'SET_CAROUSEL_ORDER';
+
+export const setCarouselOrder = ({ name, random = false, sections }) => {
+  if (!name) {
+    throw new Error('name is required');
+  }
+  if (!sections) {
+    throw new Error('sections are required');
+  }
+
+  return {
+    payload: { name, random, sections },
+    type: SET_CAROUSEL_ORDER,
+  };
+};
+
+export const initialState = {};
+
+export default function randomOrderReducer(state = initialState, action) {
+  switch (action.type) {
+    case SET_CAROUSEL_ORDER: {
+      const { name, random, sections } = action.payload;
+      const orderArray = [...Array(sections.length).keys()].slice(0, MAX_ITEMS);
```

Comment:
Could this just be `[...sections.keys()].slice(0, MAX_ITEMS);` ?

Reply:
No, because I want array indexes, not the actual content itself. The content doesn't get stored/modified in the reducer, just the order it's stored in.

Reply:
Oh snap, true. Alright, I'll fix this up. 👍